.github/workflows/workflow-executor.yml

```diff
@@ -0,0 +1,73 @@
+name: Reusable Workflow executing Makefile to execute the CI pipeline for workflows
+
+on:
+  workflow_call:
+    inputs:
+      workflow_id:
+        required: true
+        type: string
+      push_pr:
+        required: false
+        type: boolean
+        default: true
+
+env:
+  DEPLOYMENT_REPO: parodos-dev/serverless-workflows-helm
+  REGISTRY_REPO: orchestrator
+  GH_TOKEN: ${{ secrets.HELM_REPO_TOKEN }}
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build
+      run: |
+        make WORKFLOW_ID=${{ inputs.workflow_id }} \
+          REGISTRY_REPO=${{ env.REGISTRY_REPO }} \
+          REGISTRY_USERNAME=${{ secrets.NEW_QUAY_USERNAME }} \
+          REGISTRY_PASSWORD=${{ secrets.NEW_QUAY_PASSWORD }} \
+          build-image push-image
+  gen-and-push-manifests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Push manifests to CD repo
+      run: |
+        # TODO: update PR_OR_COMMIT_URL with PR's URL, if it comes from a PR
+        if [ "${{ github.event.head_commit.url }}" != "" ]; then
+          PR_OR_COMMIT_URL="${{ github.event.head_commit.url }}"
+        else
+          PR_OR_COMMIT_URL="${GITHUB_SHA::7}"
+        fi
+
+        WORKDIR=${{ runner.temp }}/serverless-workflows
+        make WORKFLOW_ID=${{ inputs.workflow_id }} \
+          WORKDIR=${WORKDIR} \
+          GIT_USER_NAME=${{ github.actor }} \
+          GIT_USER_EMAIL="${{ github.actor }}@users.noreply.github.com" \
+          GIT_TOKEN=${{ env.GH_TOKEN }} \
+          PR_OR_COMMIT_URL=${PR_OR_COMMIT_URL} \
+          gen-manifests push-manifests
+    - name: Send PR
+      run: |
+        WORKDIR=${{ runner.temp }}/serverless-workflows
+        cd ${WORKDIR}/helm-repo
+        # TODO Put these in env variables
```

Comment:
do we want to merge with this TODO and have a follow-up PR? Or at least a jira ticket to not loose track of it (same for all TODOs)

Reply:
https://github.com/parodos-dev/serverless-workflows/issues/109

Reply:
perfection